cmsBuild

```diff
@@ -3629,6 +3637,17 @@ def build(opts, args, factory):
             finalString = "\n".join(["%s (%s)" % (dep.pkgName(), dep.checksum)
                                      for dep in getPackages(deps)])
             log(finalString)
+            #get the deps for all packages to be built
+            if opts.depsLog:
+                deplist = getPackages(deps)
+                deps_object = []
+                for pkgdep in deplist:
+                    ptchs = [os.path.join(cmsdist_gh_url ,cmsdist_branch ,os.path.basename(p)) for p in pkgdep.patches]
+                    docs = os.path.join(cmsdist_gh_url, cmsdist_branch, pkgdep.name,"docs.txt")
+                    deps_object.append({"pkgname":pkgdep.name, "version_short":pkgdep.realVersion, "version_long":pkgdep.version, "checksum": pkgdep.checksum, "patches":ptchs, "sources":pkgdep.sources, "docs":docs })
```

Comment:
this is also wrong. For a package `a` which depends on package `b` and `c` then generated json file should have something like
```
{
  "name": "a",
  "version": "v-foo",
  "realversion": "v"
  "sources": ["src1","src2"],
  "patches:" ["patch1","patch2","patch3"],
  "doc": "a/doc.txt",
  "dependencies": ["b", "c"],
  "checksum": "checksum"
}
```

Reply:
I'll post an example if needed but my validation looked fine.
for example:

cmssw-dqm-reference-deployer:

```
{                                                                                                                                                                                  
  "cmssw-dqm-reference-deployer": {                                                                                                                                                
    "checksum": "4e9cf366cc4ece067e506f6a4cd9dfcf",                                                                                                                                
    "dependencies": [],                                                                                                                                                            
    "doc": "cmssw-dqm-reference-deployer/doc.txt",                                                                                                                                 
    "patches": [],                                                                                                                                                                 
    "pkgname": "cmssw-dqm-reference-deployer",                                                                                                                                     
    "realversion": "CMSSW_6_0_0_pre11",                                                                                                                                            
    "sources": [                                                                                                                                                                   
      "http://cmsrep.cern.ch/cmssw/cms/RPMS/slc5_amd64_gcc462/cms+cmssw-dqm-reference-hi+CMSSW_6_0_0_pre11-1-1.slc5_amd64_gcc462.rpm",                                             
      "http://cmsrep.cern.ch/cmssw/cms/RPMS/slc5_amd64_gcc462/cms+cmssw-dqm-reference-nonhi+CMSSW_6_0_0_pre11-1-1.slc5_amd64_gcc462.rpm"                                           
    ],                                                                                                                                                                             
    "version": "CMSSW_6_0_0_pre11"                                                                                                                                                 
  }                                                                                                                                                                                
}
```
or another example , ROOT

Reply:
{
  "OpenBLAS": {
    "checksum": "2a64f3fadf911083b652fe46a10f8236",
    "dependencies": [
      "gcc"
    ],
    "doc": "OpenBLAS/doc.txt",
    "patches": [],
    "pkgname": "OpenBLAS",
    "realversion": "0.3.9",
    "sources": [
      "https://github.com/xianyi/OpenBLAS/archive/v0.3.9.tar.gz"
    ],
    "version": "0.3.9"
  },
  "bz2lib": {
    "checksum": "bad2902e63a0e3c3f83b7c8af4fde5ba",
    "dependencies": [
      "gcc"
    ],
    "doc": "bz2lib/doc.txt",
    "patches": [],
    "pkgname": "bz2lib",
    "realversion": "1.0.6",
    "sources": [
      "http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz"
    ],
    "version": "1.0.6-bcolbf2"
  },
  "cmake": {
    "checksum": "1817fecdf60c9bf23e5a88d5a44598f6",
    "dependencies": [
      "bz2lib",
      "curl",
      "expat",
      "gcc",
      "openssl",
      "zlib"
    ],
    "doc": "cmake/doc.txt",
    "patches": [],
    "pkgname": "cmake",
    "realversion": "3.17.2",
    "sources": [
      "http://www.cmake.org/files/v3.17/cmake-3.17.2.tar.gz"
    ],
    "version": "3.17.2"
  },
  "curl": {
    "checksum": "4188e0dbbff488fa884e2c08bef5e64f",
    "dependencies": [
      "gcc",
      "openssl",
      "zlib"
    ],
    "doc": "curl/doc.txt",
    "patches": [],
    "pkgname": "curl",
    "realversion": "7.70.0",
    "sources": [
      "http://curl.haxx.se/download/curl-7.70.0.tar.gz"
    ],
    "version": "7.70.0"
  },
  "davix": {
    "checksum": "ce9dead53503dda4161008e3128585a3",
    "dependencies": [
      "autotools",
      "bz2lib",
      "cmake",
      "curl",
      "expat",
      "gcc",
      "git",
      "gmake",
      "libuuid",
      "libxml2",
      "openssl",
      "pcre",
      "xz",
      "zlib"
    ],
    "doc": "davix/doc.txt",
    "patches": [],
    "pkgname": "davix",
    "realversion": "0.7.3",
    "sources": [
      "git+https://github.com/cern-it-sdc-id/davix.git?obj=master/R_0_7_3&export=davix-0.7.3&output=/davix-0.7.3.tgz"
    ],
    "version": "0.7.3-ghbfee"
  },
  "db6": {
    "checksum": "a2b0bb54fa02cd3c0b0b079c7e6df2da",
    "dependencies": [
      "gcc"
    ],
    "doc": "db6/doc.txt",
    "patches": [],
    "pkgname": "db6",
    "realversion": "6.2.32",
    "sources": [
      "http://cmsrep.cern.ch/cmssw/download/db-6.2.32.tar.gz"
    ],
    "version": "6.2.32-bcolbf"
  },
  "dcap": {
    "checksum": "9277ef1ad1cc90db6224a4ac6980753c",
    "dependencies": [
      "autotools",
      "gcc",
      "zlib"
    ],
    "doc": "dcap/doc.txt",
    "patches": [],
    "pkgname": "dcap",
    "realversion": "2.47.12",
    "sources": [
      "git+https://github.com/cms-externals/dcap.git?obj=cms/2.47.12/5753eec777a47908a40de670094903ce6b13176b&export=dcap-2.47.12&output=/dcap-2.47.12-5753eec777a47908a40de670094903ce6b13176b.tgz"
    ],
    "version": "2.47.12-bcolbf2"
  },
  "expat": {
    "checksum": "7e999e629861a96815c1dc71fb72f8c5",
    "dependencies": [
      "gcc"
    ],
    "doc": "expat/doc.txt",
    "patches": [],
    "pkgname": "expat",
    "realversion": "2.1.0",
    "sources": [
      "http://downloads.sourceforge.net/project/expat/expat/2.1.0/expat-2.1.0.tar.gz"
    ],
    "version": "2.1.0-bcolbf2"
  },
  "fftw3": {
    "checksum": "fdd70fbe5e8b6639da1e0199b761d5cd",
    "dependencies": [
      "gcc"
    ],
    "doc": "fftw3/doc.txt",
    "patches": [],
    "pkgname": "fftw3",
    "realversion": "3.3.8",
    "sources": [
      "http://www.fftw.org/fftw-3.3.8.tar.gz"
    ],
    "version": "3.3.8-bcolbf"
  },
  "freetype": {
    "checksum": "e65f2891acd95dc7b96b866f772dc09b",
    "dependencies": [
      "bz2lib",
      "gcc",
      "libpng",
      "zlib"
    ],
    "doc": "freetype/doc.txt",
    "patches": [],
    "pkgname": "freetype",
    "realversion": "2.10.0",
    "sources": [
      "http://download.savannah.gnu.org/releases/freetype/freetype-2.10.0.tar.bz2"
    ],
    "version": "2.10.0-bcolbf2"
  },
  "gcc": {
    "checksum": "a7ef4723f4150ff274bca74f8bfab899",
    "dependencies": [],
    "doc": "gcc/doc.txt",
    "patches": [
      "gcc-flex-nonfull-path-m4.patch",
      "gcc-flex-disable-doc.patch"
    ],
    "pkgname": "gcc",
    "realversion": "8.2.0",
    "sources": [
      "git+https://github.com/gcc-mirror/gcc.git?obj=releases/gcc-8/8cd3bffead2ed1d1998c190865694f920fbc93ab&export=gcc-8.2.0&output=/gcc-8.2.0-8cd3bffead2ed1d1998c190865694f920fbc93ab.tgz",
      "https://gmplib.org/download/gmp/gmp-6.2.0.tar.bz2",
      "http://www.mpfr.org/mpfr-4.0.2/mpfr-4.0.2.tar.bz2",
      "https://ftp.gnu.org/gnu/mpc/mpc-1.1.0.tar.gz",
      "http://isl.gforge.inria.fr/isl-0.21.tar.bz2",
      "http://zlib.net/zlib-1.2.11.tar.gz",
      "http://ftp.gnu.org/gnu/bison/bison-3.5.tar.gz",
      "https://sourceware.org/pub/binutils/releases/binutils-2.32.tar.bz2",
      "https://sourceware.org/pub/elfutils/0.177/elfutils-0.177.tar.bz2",
      "http://ftp.gnu.org/gnu/m4/m4-1.4.18.tar.gz",
      "https://github.com/westes/flex/releases/download/v2.6.4/flex-2.6.4.tar.gz"
    ],
    "version": "8.2.0-bcolbf"
  },
  "gdbm": {
    "checksum": "f2837e7c35183a675e344714418de797",
    "dependencies": [
      "gcc"
    ],
    "doc": "gdbm/doc.txt",
    "patches": [],
    "pkgname": "gdbm",
    "realversion": "1.10",
    "sources": [
      "http://ftp.gnu.org/gnu/gdbm/gdbm-1.10.tar.gz"
    ],
    "version": "1.10-bcolbf2"
  },
  "giflib": {
    "checksum": "134c7d87aba101a1fc1159bae797ac79",
    "dependencies": [
      "autotools",
      "gcc"
    ],
    "doc": "giflib/doc.txt",
    "patches": [],
    "pkgname": "giflib",
    "realversion": "5.2.0",
    "sources": [
      "https://sourceforge.net/projects/giflib/files/giflib-5.2.0.tar.gz"
    ],
    "version": "5.2.0-bcolbf2"
  },
  "gsl": {
    "checksum": "2233567d64630d2ca8fc8c766cccb0e1",
    "dependencies": [
      "OpenBLAS",
      "gcc"
    ],
    "doc": "gsl/doc.txt",
    "patches": [],
    "pkgname": "gsl",
    "realversion": "2.6",
    "sources": [
      "ftp://ftp.gnu.org/gnu/gsl/gsl-2.6.tar.gz"
    ],
    "version": "2.6-bcolbf3"
  },
  "libffi": {
    "checksum": "959d74d36cc1ec509d77d3e5de6225cf",
    "dependencies": [
      "gcc"
    ],
    "doc": "libffi/doc.txt",
    "patches": [
      "libffi-3.2.1-fix-include-path.patch"
    ],
    "pkgname": "libffi",
    "realversion": "3.2.1",
    "sources": [
      "ftp://sourceware.org/pub/libffi/libffi-3.2.1.tar.gz"
    ],
    "version": "3.2.1-bcolbf"
  },
  "libjpeg-turbo": {
    "checksum": "4f6bd94252cf916a6c6ce858175c7dc0",
    "dependencies": [
      "autotools",
      "bz2lib",
      "cmake",
      "curl",
      "expat",
      "gcc",
      "gmake",
      "nasm",
      "openssl",
      "zlib"
    ],
    "doc": "libjpeg-turbo/doc.txt",
    "patches": [],
    "pkgname": "libjpeg-turbo",
    "realversion": "2.0.2",
    "sources": [
      "https://github.com/libjpeg-turbo/libjpeg-turbo/archive/2.0.2.tar.gz"
    ],
    "version": "2.0.2-ghbfee"
  },
  "libpng": {
    "checksum": "7d1c2bd4a26954d23f0c4db374df597d",
    "dependencies": [
      "autotools",
      "gcc",
      "zlib"
    ],
    "doc": "libpng/doc.txt",
    "patches": [],
    "pkgname": "libpng",
    "realversion": "1.6.37",
    "sources": [
      "git+https://github.com/cms-externals/libpng.git?obj=cms/v1.6.37/a40189cf881e9f0db80511c382292a5604c3c3d1&export=libpng-1.6.37&output=/libpng-1.6.37.tgz"
    ],
    "version": "1.6.37-bcolbf2"
  },
  "libtiff": {
    "checksum": "9936248834c6d54398f7739092da72ab",
    "dependencies": [
      "gcc",
      "libjpeg-turbo",
      "zlib"
    ],
    "doc": "libtiff/doc.txt",
    "patches": [],
    "pkgname": "libtiff",
    "realversion": "4.0.10",
    "sources": [
      "http://download.osgeo.org/libtiff/tiff-4.0.10.zip"
    ],
    "version": "4.0.10-ghbfee"
  },
  "libuuid": {
    "checksum": "7539a044e9020247154b225efb4154e7",
    "dependencies": [
      "gcc"
    ],
    "doc": "libuuid/doc.txt",
    "patches": [
      "libuuid-2.34-disable-get_uuid_via_daemon.patch"
    ],
    "pkgname": "libuuid",
    "realversion": "2.34",
    "sources": [
      "http://www.kernel.org/pub/linux/utils/util-linux/v2.34/util-linux-2.34.tar.gz"
    ],
    "version": "2.34-bcolbf2"
  },
  "libxml2": {
    "checksum": "6ade7e27e7a7edff65a55dfe499d9393",
    "dependencies": [
      "gcc",
      "xz",
      "zlib"
    ],
    "doc": "libxml2/doc.txt",
    "patches": [],
    "pkgname": "libxml2",
    "realversion": "2.9.10",
    "sources": [
      "ftp://xmlsoft.org/libxml2/libxml2-2.9.10.tar.gz"
    ],
    "version": "2.9.10"
  },
  "lz4": {
    "checksum": "eea05402db539b04dc026ac2c0ca12aa",
    "dependencies": [
      "gcc"
    ],
    "doc": "lz4/doc.txt",
    "patches": [],
    "pkgname": "lz4",
    "realversion": "1.9.2",
    "sources": [
      "https://github.com/lz4/lz4/archive/v1.9.2.tar.gz"
    ],
    "version": "1.9.2"
  },
  "ninja": {
    "checksum": "7e80468514b9e9f20f6a8ca21b76d598",
    "dependencies": [
      "bz2lib",
      "db6",
      "expat",
      "gcc",
      "gdbm",
      "libffi",
      "openssl",
      "python",
      "re2c",
      "sqlite",
      "zlib"
    ],
    "doc": "ninja/doc.txt",
    "patches": [],
    "pkgname": "ninja",
    "realversion": "1.10.0",
    "sources": [
      "git://github.com/ninja-build/ninja.git?obj=release/v1.10.0&export=ninja-1.10.0&output=/ninja-1.10.0.tgz"
    ],
    "version": "1.10.0"
  },
  "openssl": {
    "checksum": "9b79bec29f96c8347b7a6da014a6d675",
    "dependencies": [
      "gcc"
    ],
    "doc": "openssl/doc.txt",
    "patches": [
      "openssl-1.0.2d-pr3979.patch",
      "openssl-1.0.2d-disable-install-openssldir.patch"
    ],
    "pkgname": "openssl",
    "realversion": "1.0.2d",
    "sources": [
      "http://davidlt.web.cern.ch/davidlt/vault/openssl-1.0.2d-5675d07a144aa1a6c85f488a95aeea7854e86059.tar.bz2"
    ],
    "version": "1.0.2d-bcolbf"
  },
  "pcre": {
    "checksum": "010ebbb92b45aac1ddcc311fc4d42321",
    "dependencies": [
      "bz2lib",
      "gcc",
      "zlib"
    ],
    "doc": "pcre/doc.txt",
    "patches": [],
    "pkgname": "pcre",
    "realversion": "8.43",
    "sources": [
      "http://downloads.sourceforge.net/pcre/pcre-8.43.tar.bz2"
    ],
    "version": "8.43-bcolbf"
  },
  "py2-numpy": {
    "checksum": "e8de7410dfd2f04bba99e6f3800ed589",
    "dependencies": [
      "OpenBLAS",
      "bz2lib",
      "db6",
      "expat",
      "gcc",
      "gdbm",
      "libffi",
      "libuuid",
      "openssl",
      "py2-pip",
      "py2-setuptools",
      "python",
      "python3",
      "sqlite",
      "xz",
      "zlib"
    ],
    "doc": "py2-numpy/doc.txt",
    "patches": [],
    "pkgname": "py2-numpy",
    "realversion": "1.16.6",
    "sources": [
      "https://github.com/numpy/numpy/releases/download/v1.16.6/numpy-1.16.6.tar.gz"
    ],
    "version": "1.16.6-bcolbf"
  },
  "py2-setuptools": {
    "checksum": "fe7d6189aeb3ee770097a49e2b28c50f",
    "dependencies": [
      "bz2lib",
      "db6",
      "expat",
      "gcc",
      "gdbm",
      "libffi",
      "libuuid",
      "openssl",
      "python",
      "python3",
      "sqlite",
      "xz",
      "zlib"
    ],
    "doc": "py2-setuptools/doc.txt",
    "patches": [],
    "pkgname": "py2-setuptools",
    "realversion": "41.2.0",
    "sources": [
      "https://github.com/pypa/setuptools/archive/v41.2.0.tar.gz"
    ],
    "version": "41.2.0-bcolbf2"
  },
  "python": {
    "checksum": "72f7bc9554e235317940701eefc58b3f",
    "dependencies": [
      "bz2lib",
      "db6",
      "expat",
      "gcc",
      "gdbm",
      "libffi",
      "openssl",
      "sqlite",
      "zlib"
    ],
    "doc": "python/doc.txt",
    "patches": [],
    "pkgname": "python",
    "realversion": "2.7.15",
    "sources": [
      "git+https://github.com/cms-externals/cpython.git?obj=cms/2.7/c40eeeb/583c0f2237aade97173702ac602b16f51eadde6d&export=python-2.7.15&output=/python-2.7.15.tgz",
      "/build/mrodozov/deps/cmsdist/valgrind-python.supp.file"
    ],
    "version": "2.7.15-bcolbf2"
  },
  "python3": {
    "checksum": "540b446afc2666202950081e91cd5259",
    "dependencies": [
      "bz2lib",
      "db6",
      "expat",
      "gcc",
      "gdbm",
      "libffi",
      "libuuid",
      "openssl",
      "sqlite",
      "xz",
      "zlib"
    ],
    "doc": "python3/doc.txt",
    "patches": [],
    "pkgname": "python3",
    "realversion": "3.8.2",
    "sources": [
      "https://www.python.org/ftp/python/3.8.2/Python-3.8.2.tgz"
    ],
    "version": "3.8.2-bcolbf"
  },
  "re2c": {
    "checksum": "59eed36cb4212389ca617ea71ea51824",
    "dependencies": [
      "gcc"
    ],
    "doc": "re2c/doc.txt",
    "patches": [],
    "pkgname": "re2c",
    "realversion": "0.13.5",
    "sources": [
      "http://heanet.dl.sourceforge.net/project/re2c/re2c/0.13.5/re2c-0.13.5.tar.gz"
    ],
    "version": "0.13.5-bcolbf"
  },
  "root": {
    "checksum": "738e64c5acf441d1eace3bb20e26bd11",
    "dependencies": [
      "OpenBLAS",
      "bz2lib",
      "cmake",
      "curl",
      "davix",
      "db6",
      "dcap",
      "expat",
      "fftw3",
      "freetype",
      "gcc",
      "gdbm",
      "giflib",
      "gsl",
      "libffi",
      "libjpeg-turbo",
      "libpng",
      "libtiff",
      "libuuid",
      "libxml2",
      "lz4",
      "ninja",
      "openssl",
      "pcre",
      "py2-numpy",
      "py2-setuptools",
      "python",
      "python3",
      "re2c",
      "sqlite",
      "tbb",
      "xrootd",
      "xz",
      "zlib"
    ],
    "doc": "root/doc.txt",
    "patches": [],
    "pkgname": "root",
    "realversion": "6.18.04",
    "sources": [
      "git+https://github.com/cms-sw/root.git?obj=cms/v6-18-00-patches/v6-18-04/2b91a0f8b8a094137a90fed9b340d99a3108c2d4&export=root-6.18.04&output=/root-6.18.04-2b91a0f8b8a094137a90fed9b340d99a3108c2d4.tgz"
    ],
    "version": "6.18.04-ghbfee"
  },
  "sqlite": {
    "checksum": "b9686715a59b2465d1c05e7942b26c18",
    "dependencies": [
      "gcc"
    ],
    "doc": "sqlite/doc.txt",
    "patches": [],
    "pkgname": "sqlite",
    "realversion": "3.22.0",
    "sources": [
      "https://www.sqlite.org/2018/sqlite-autoconf-3220000.tar.gz"
    ],
    "version": "3.22.0-bcolbf"
  },
  "tbb": {
    "checksum": "a1c6d1ac5ee19d76ca7b6ad9118c7586",
    "dependencies": [
      "gcc"
    ],
    "doc": "tbb/doc.txt",
    "patches": [],
    "pkgname": "tbb",
    "realversion": "2019_U9",
    "sources": [
      "git+https://github.com/01org/tbb.git?obj=tbb_2019/2019_U9&export=tbb-2019_U9&output=/tbb-2019_U9-2019_U9.tgz"
    ],
    "version": "2019_U9-bcolbf"
  },
  "xrootd": {
    "checksum": "23c562fc4e34414396ec03b8458929ec",
    "dependencies": [
      "bz2lib",
      "cmake",
      "curl",
      "db6",
      "expat",
      "gcc",
      "gdbm",
      "libffi",
      "libuuid",
      "libxml2",
      "openssl",
      "python",
      "python3",
      "sqlite",
      "xz",
      "zlib"
    ],
    "doc": "xrootd/doc.txt",
    "patches": [],
    "pkgname": "xrootd",
    "realversion": "4.10.0",
    "sources": [
      "git+https://github.com/cms-externals/xrootd.git?obj=cms/v4.10.0/9d3b80504a2a142723038f207c3e950347c756aa&export=xrootd-4.10.0&output=/xrootd-4.10.0.tgz"
    ],
    "version": "4.10.0-ghbfee"
  },
  "xz": {
    "checksum": "201073c153b33cecf7f7e4676a2513ee",
    "dependencies": [
      "autotools",
      "gcc"
    ],
    "doc": "xz/doc.txt",
    "patches": [],
    "pkgname": "xz",
    "realversion": "5.2.4",
    "sources": [
      "git+https://github.com/cms-externals/xz.git?obj=cms/v5.2.4/982e837ddd3da234f710870fa65ba8c14f381b8b&export=xz-5.2.4&output=/xz-5.2.4.tgz"
    ],
    "version": "5.2.4-bcolbf2"
  },
  "zlib": {
    "checksum": "31e7726de95f47cbeb14cc45444f56d0",
    "dependencies": [
      "gcc"
    ],
    "doc": "zlib/doc.txt",
    "patches": [],
    "pkgname": "zlib",
    "realversion": "1.2.11",
    "sources": [
      "git://github.com/cms-externals/zlib.git?obj=cms/v1.2.11/822f7f5a8c57802faf8bbfe16266be02eff8c2e2&export=zlib-1.2.11&output=/zlib-1.2.11.tgz"
    ],
    "version": "1.2.11-bcolbf"
  }
}